code/game/machinery/atmos_exporter.dm

```diff
@@ -0,0 +1,278 @@
+						C.anchored = 1
+						for(var/obj/machinery/atmos_points_exporter/EX in machines)
+							EX.ship()
+						for(var/obj/machinery/engi_points_manager/EPM in machines)
+							EPM.GBP += price
+							EPM.GBPearned += price
+						if(export_target == "premium")
+							visible_message("<span class='danger'>You have been blessed by the atmos gods for exporting a premium canister!</span>")
+							var/prize = pick(/obj/item/clothing/under/rank/atmos_elite,/obj/item/clothing/head/atmos_hood,/obj/item/clothing/neck/cloak/atmos)
+							new /obj/effect/overlay/temp/explosion/fast(get_turf(src))
+							new prize(get_turf(src))
+							playsound(src, 'sound/effects/pray_chaplain.ogg', 100, 1)
+						quality.Remove(export_target)
+						reset()
+					else
+						visible_message("<span class='danger'>Invalid temperature detected!</span>")
+						playsound(src, 'sound/machines/defib_failed.ogg', 100, 1)
+				else
+					visible_message("<span class='danger'>Invalid pressure detected!</span>")
+					playsound(src, 'sound/machines/defib_failed.ogg', 100, 1)
+
+			src.updateUsrDialog()
+
+/obj/machinery/atmos_points/proc/reset()
+	export_name = "------"
+	export_target = ""
+	gas.Cut()
+	concentration.Cut()
+	upressure = 0
+	lpressure = 0
+	utemp = 0
+	ltemp = 0
+
+/obj/machinery/atmos_points/process()
+	if(market_cooldown < world.time)
+		reset()
+		economy_name = corp_name()
+		e_gas.Cut()
+		var/list/gas_list = list("o2","n2","n2o","plasma","co2")
+		var/e_choice = rand(1,5)
+		e_gas += gas_list[e_choice]
+		e_pressure = rand(500,4500)
+		e_temp = rand(75,550)
+		e_concentration1 = 100
+		e_price = rand(200,500)
+
+		standard_name = corp_name()
+		s_gas.Cut()
+		gas_list = list("o2","n2","n2o","plasma","co2")
+		for(var/x in 1 to 2)
+			var/s_choice = rand(1,6-x)
+			s_gas += gas_list[s_choice]
+			gas_list -= gas_list[s_choice]
+		s_pressure = rand(500,4500)
+		s_temp = rand(75,550)
+		s_concentration1 = rand(60,90)
+		s_concentration2 = 100 - s_concentration1
+		s_price = rand(500,950)
+
+		premium_name = corp_name()
+		p_gas.Cut()
+		gas_list = list("o2","n2","n2o","plasma","co2")
+		for(var/y in 1 to 3)
+			var/p_choice = rand(1,6-y)
+			p_gas += gas_list[p_choice]
+			gas_list -= gas_list[p_choice]
+		p_pressure = rand(3000,9000)
+		p_temp = rand(0,1)
+		if(p_temp)
+			p_temp = rand(600,8000)
+		else
+			p_temp = rand(5,70)
+		p_concentration1 = rand(10,45)
+		p_concentration2 = rand(10,45)
+		p_concentration3 = 100 - p_concentration1 - p_concentration2
+		p_price = rand(1500,3000)
+
+		market_cooldown = world.time + 3000
+		quality = list("Economy","Standard","Premium")
+
+
+
+/obj/machinery/atmos_points/proc/corp_name()
+	var/name = ""
+
+	// Prefix
+	name += pick("Clandestine", "Prima", "Blue Sigil", "Front", "Solar Mass", "Atmosia", "Shell", "North", "Omni", "Newton", "Cyber", "Red Harvest", "Gene", "Plasmatech", "Robustin", "Honkorp", "Scarborough", "Terminus", "Veritas", "Dark Mind", "Overmind", "Singulo", "Viva", "Grey Matter")

```

Comment:
So I'm just basically moving it to a different file? I'm really not in the mood to rewrite this proc so that it works more like Ion laws and becomes a 500 line switch list.